### DISCUSSION TITLE
Raspberry pi 5/gopro 10 usb ethernet not available after several poweroff/on cycles

### DISCUSSION THREAD

**Discussion by hkskoglund** [gopro/labs#875] [Q&A] [answered]
Hi! I am testing an raspberry pi 5 and gopro 10 for a photo timelapse project. I found that using usb ethernet is faster than 5Ghz wifi for downloads, so I wanted test it. I had to set usb_max_current_enable=1 in rpi5 /boot/firmware/config.txt to get sufficient power.

I first tried to use stock firmware and always got the "NOT ENOUGH POWER" warning after raspberry pi bootup. I found that physically reinserting the gopro usb-cable boots up the gopro in "USB CONNECTED" and usb-ethernet mode.  And everything seems to works ok even on just 1.6A.

I found that using labs firmware with !MTUSB=1 QR-code boots up the gopro without reinserting the cable. I have tested  manually power on/power off cycling many times. Eventually the gopro only boots up in normal UI without usb ethernet (features activated TUSB/beta terms screens). Then I have to reinsert the usb cable to get "USB Connected" ui/ethernet again. 

I dont known if this is a pi 5 issue or a gopro issue?

**Comment by dnewman-gpsw** [comment, marked as answer]
I don't know. Labs doesn't modify this behavior, but I investigating whether a Labs control can be added as there is a HERO10 release in the works.

**Comment by AthenasArch** [comment]
[hkskoglund](https://github.com/hkskoglund),

Did you manage to perform the timelapse process with the Gopro with the Raspberry pi 5?

I'm working on a similar project, where I need to connect the GoPro Hero 12 via USB to my Raspberry Pi5 and collect high quality images every second.

I'm having difficulties, do you have any recommendations?

The camera must be turned on when turning on the Raspberry Pi, I'm using GoPro Labs for some tasks, but I'm having difficulty collecting the images on Ubuntu.

Do you have any suggestions on which way to go?

**Reply by hkskoglund** [reply]
As far as I can conclude from testing, for the gopro 10 its not possible not wake it up via bluetooth and enter usb ethernet mode for fast file transfer. USB ethernet is only available when pushing power on button. There is no bluetooth command to enter usb ethernet mode. I don't known about the usb ethernet features available on gopro 12. I found usefull information on https://gopro.github.io/OpenGoPro/
